internal/database/postgres/postgres_test.go

```diff
@@ -0,0 +1,43 @@
+package postgres
+
+import (
+	"os"
+	"testing"
+
+	"github.com/gpuctl/gpuctl/internal/database"
+)
+
+// run all the database unit tests on the postgres implementation
+func TestPostgres(t *testing.T) {
+	// set default value that matches github workflow
+	url := os.Getenv("TEST_URL")
+	if url == "" {
+		url = "postgres://postgres@localhost/postgres"
+	}
+
+	for _, test := range database.UnitTests {
```

Comment:
Actually I would suggest that you wrap the entire test in a transaction rather than create/drop tables.

Reply:
I don't think that would be doable (at least working within the confines of the go sql library). The postgres implementation already wraps most of its work in transactions, so executing tests within a transaction would require nested transactions, and the transaction type [`sql.Tx`](https://pkg.go.dev/database/sql#Tx) doesn't have a function for creating transactions within it.

In theory nested transactions look like they'd be possible using [savepoints](https://stackoverflow.com/questions/25425944/does-postgres-support-nested-or-autonomous-transactions), but I'm not sure how we could elegantly use savepoints whilst testing, but then go back to using transactions during production?

Reply:
It would be great if possible though, because then we could use Postgres' temporary tables (tables that're only visible within one transaction and destroyed after) and avoid a whole host of issues that currently can occur if you run `go test ./...` in more than one terminal at a time

Reply:
Ah that's annoying, I suppose we'll have to keep on the look out for that breaking then.